File: Easy/1266. Minimum Time Visiting All Points.java
class Solution {
    public int minTimeToVisitAllPoints(int[][] points) {
        int ans = 0;
        for (int i = 1; i < points.length; i++) {
            int[] A = points[i - 1], B = points[i];
            int H = Math.abs(A[0] - B[0]);
            int V = Math.abs(A[1] - B[1]);
            int D = Math.min(H, V);
            ans += (H + V - D);
        }
        return ans;
    }
}

File: Easy/1266. Minimum Time Visiting All Points.py
from typing import List


class Solution:
    def minTimeToVisitAllPoints(self, points: List[List[int]]) -> int:
        result = 0
        prev_x, prev_y = points[0][0], points[0][1]
        for curr_x, curr_y in points[1:]:
            result += max(abs(prev_x - curr_x), abs(prev_y - curr_y))
            prev_x, prev_y = curr_x, curr_y
        return result